specification/compiler/steps/validate-model.ts

```diff
@@ -139,6 +139,18 @@ export default async function validateModel (apiModel: model.Model): Promise<mod
   // Used both to avoid re-validating types we already visited and to find dangling types at the end.
   const typesSeen = new Set<string>()
 
+  // The ErrorResponse is not referenced anywhere, but we need to export it
+  // as any API could return it if an error happens. It's widely used in
+  // the type test as well.
+  let ErrorResponseType: model.Interface | undefined
+  for (const type of apiModel.types) {
+    if (type.name.name === 'ErrorResponse') {
+      ErrorResponseType = type as model.Interface
+      break
+    }
+  }
+
```

Comment:
You can replace all this (and the `push` below) with:

```ts
validateTypeRef({namespace: "common_abstractions.response", name: "ErrorResponse"}, undefined, new Set());
```

This will have the added benefits of validating this type definition instead of just adding it to the list of visisted types.